tools/jenkins/restapi_common.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# This common file writes the functions of getting qps from flink restful api
+#
+# Note by AiHua Li:
+
+import urllib2
+import json
+
+
+def execute_get(url):
+    result_url = ""
+    try:
+        result_url = urllib2.urlopen(url, timeout=300).read()
+    except urllib2.URLError, e:
+        print e.reason
+    return result_url
+
+
+def get_source_node(plan):
+    source_nodes = []
+    nodes = plan["nodes"] if "nodes" in plan else []
+    for node in nodes:
+        inputs = node["inputs"] if "inputs" in node else []
+        if len(inputs) == 0:
```

Comment:
maybe here we want to append to `source_nodes` when `len(inputs)` is _not_ zero?